src/Plugin/RulesAction/SystemMailToUsersOfRole.php

```diff
@@ -0,0 +1,169 @@
+ *       required = FALSE
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class SystemMailToUsersOfRole extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * @var \Drupal\Core\Mail\MailManagerInterface
+   */
+  protected $mailManager;
+
+  /**
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * @var \Drupal\user\UserStorage
+   */
+  protected $userStorage;
+
+  /**
+   * Constructs a SendMailToUsersOfRole object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The storage service.
+   * @param \Drupal\Core\Mail\MailManagerInterface $mail_manager
+   *   The mail manager service.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\user\UserStorage $userStorage
+   *   The user storage service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LoggerInterface $logger, MailManagerInterface $mail_manager, ConfigFactoryInterface $config_factory, UserStorage $userStorage) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->logger = $logger;
+    $this->mailManager = $mail_manager;
+    $this->config = $config_factory->get('site.config');
+    $this->userStorage = $userStorage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.factory'),
+      $container->get('plugin.manager.mail'),
+      $container->get('config.factory'),
+      $container->get('entity.manager')->getStorage('user')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    return $this->t('Sends an e-mail to the users of a role');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute() {
+    $roles = $this->getContextValue('roles');
+
+    if (empty($roles)) {
+      return;
+    }
+    // Get the role ids and remove the empty ones (in case for example the role
+    // has been removed in the meantime).
+    $rids = array_filter(array_map(function ($role) {
+      return $role->id();
+    }, $roles));
+    if (empty($rids)) {
+      return;
+    }
+
+    // Get now all the users that match the roles (at least one of the role).
+    $accounts = $this->userStorage
+      ->loadByProperties(['roles' => $roles]);
+    // @todo: Should we implement support for user tokens in subject and body? In the
```

Comment:
This todo should be removed. We handle token replacements outside actions, so we are good here.